frontend/packages/topology/src/components/quick-search/QuickSearchList.tsx

```diff
@@ -0,0 +1,117 @@
+import * as React from 'react';
+import { useTranslation } from 'react-i18next';
+import cx from 'classnames';
+import { history, removeQueryArgument } from '@console/internal/components/utils';
+import { CatalogItem } from '@console/plugin-sdk';
+import {
+  DataList,
+  DataListItem,
+  DataListItemRow,
+  DataListItemCells,
+  DataListCell,
+  Split,
+  SplitItem,
+  Label,
+  TextContent,
+  Text,
+  TextVariants,
+  Button,
+} from '@patternfly/react-core';
+import './QuickSearchList.scss';
+
+interface QuickSearchListProps {
+  listItems: CatalogItem[];
+  totalItems: number;
+  selectedItemId: string;
+  searchTerm: string;
+  namespace: string;
+  onSelectListItem: (itemId: string) => void;
+}
+
+const QuickSearchList: React.FC<QuickSearchListProps> = ({
+  listItems,
+  totalItems,
+  selectedItemId,
+  searchTerm,
+  namespace,
+  onSelectListItem,
+}) => {
+  const { t } = useTranslation();
+  const ref = React.useRef(null);
+
+  const openForm = (e: React.SyntheticEvent) => {
+    e.persist();
+    e.stopPropagation();
+    e.preventDefault();
+    const targetItemUID = ref?.current?.props?.id;
+    const item = listItems.find((listItem) => listItem.uid === targetItemUID);
+    removeQueryArgument('catalogSearch');
+    history.push(item?.cta?.href);
```

Comment:
Hmmm... if we don't get the href we wanted here... you're going to redirect a bit wildly against an unknown/falsey value. Do we have an error state from UX instead?

Reply:
We do not have an error state for this as of now but for the time being what I can do is that navigate the user to the form page only when we have the href available.

Reply:
Hmmm, I wonder if fallback should be to route them just to the catalog as if they selected "all"? I know it's not ideal and would appear to be a bug... but I think it's better than `history.push(undefined)`

Reply:
Hmm yeah makes sense. @bgliwa01 It will be really helpful if you can give some input on how do we handle this on Monday.

Reply:
I suggest you do this instead so you can avoid the lookup again and the null case for item not found:
`onDoubleClick={(e) => openForm(e, item)}`

Furthermore the spec `CatalogItem` requires `cta.href` to always be defined. Make sure you aren't adding optional chaining when it isn't required.

Reply:
I don't see an error state in the developer catalog (I could be missing it) but I'm confused why we would need one here